tests/metrics/test_rest.py

```diff
@@ -43,3 +43,49 @@ async def test_rest_metrics(
     assert rest_server_requests is not None
     assert len(rest_server_requests.samples) == 1
     assert rest_server_requests.samples[0].value == expected_handled
+
+async def test_rest_queue_metrics(
```

Comment:
In order to keep the test context relevant, I would add a new `tests/metrics/test_queue.py` file which contains both `test_rest_queue_metrics` and `test_rest_batch_metrics` (which should be renamed to `test_parallel_queue_metrics` and `test_batch_queue_metrics` respectively). 

This will help maintaining these changes going forward (otherwise maintainers may find confusing what these tests are for).